packages/fiori/src/ViewSettingsDialog.js

```diff
@@ -219,6 +219,10 @@ class ViewSettingsDialog extends UI5Element {
 		if (this._currentSettings.filters && this._currentSettings.filters.length) {
 			this._setAdditionalTexts();
 		}
+
+		if (!this.shouldBuildSort) {
```

Comment:
I think we could check here if we <code>shouldBuildFilter</code> as well before setting the <code>_currentMode</code> to <code>ViewSettingsDialogMode.Filter</code>.
In the following method <code>_restoreConfirmedOnEscape</code> we'll always restore the mode to <code>ViewSettingsDialogMode.Sort</code> even if
<code>shouldBuildSort</code> is equal to <code>false</code>. So we should make the same check there too.

Reply:
The case in _restoreConfirmedOnEscape will always be handled by the onBeforeRendering logic.